### PR TITLE
Add visibility icons to statuses

### PR DESCRIPTION
**Pull request by wbrawner**
Hi! :wave:

I realize this isn't a part of standard Mastodon though I'm a fan of the features requested in #835 and the data is already there so it may as well be used, right? I'm open to any suggestions for improvements here. Screenshots below:

EDIT: I just realized that devices that have modified the system font size end up with really bad alignment with the icon/text. I'll work on fixing that but I'd appreciate any comments/concerns in the meantime :) Also CI doesn't seem to be happy so I'll fix whatever I broke :P

image
image

### CODE REVIEW

**Comment by connyduck** [comment]
Im fine with having this on the detailed status, but not in timelines. There is already not enough space for everything in this one line.

**Comment by charlag** [comment]
Same here

**Comment by wbrawner** [comment]
Yeah, it does seem a bit cluttered. I went ahead and pulled it out of the timeline statuses. I also implemented dynamic scaling of the visibility icon based on the font size of the timestamp, so that should look a lot better now on devices where users have adjusted the system font size.

**Comment by wbrawner** [comment]
I moved the logic for selecting the icon into the `setStatusVisibility()` method, since it's most likely the only place it'll be used as far as I can tell. I also had to migrate `setStatusVisibility()` into the `StatusDetailedViewHolder` class since I no longer was able to check for the existence of the `ImageView`. I used `TextView.getTextSize()` for the height of the icon, and it seems to be closer, but still off by a couple of pixels. If you have any suggestions for how to get it pixel-perfect I'd appreciate it!

**Comment by charlag** [comment]
Thanks for the changes! I'll look up how to make drawable of the same size.

**Comment by wbrawner** [comment]
I've been doing some research into this myself as well, and I think that `[EditText#getTextSize()](https://developer.android.com/reference/android/widget/TextView#getTextSize())` is as close as we'll get with the sizing of the drawable. From my tinkering around with it, it seems that the little bit of extra space on the top that the drawable extends past is still considered as the text height, as accents and umlauts extend up into that space on capitalized letters.

**Comment by charlag** [comment]
@wbrawner extra space is not a problem, I think it's as intended.
Another option would be taking sp and calculating text size by hand.
Here are some options: https://stackoverflow.com/questions/2069810/how-to-assign-text-size-in-sp-value-using-java-code

**Comment by wbrawner** [comment]
> Another option would be taking sp and calculating text size by hand

I think that might be a bit of unnecessary extra work. According to the docs for `getTextSize()`, it should already be returning the size in pixels of the text, which is what I need to give to the drawable for its size. At any rate, I'll give it a shot and see if it gives me anything different from `getTextSize()`

**Comment by charlag** [comment]
It just may be easier than setting a listener etc but honestly I don't mind, it's not a timeline.

**Comment by wbrawner** [comment]
Ah, I see. Yeah the listener was necessary to wait until the TextView had already been drawn, but I can see if the `sp` -> `px` conversion is any simpler without losing accuracy

**Comment by charlag** [comment]
I may have a look again but I think that w need decision by @connyduck

File: app/src/main/java/com/keylesspalace/tusky/adapter/StatusDetailedViewHolder.java [hunk 1]
```diff
@@ -154,6 +155,55 @@ void setupWithStatus(final StatusViewData.Concrete status, final StatusActionLis
...
+                break;
+            case DIRECT:
+                visibilityIcon = R.drawable.ic_email_24dp;
+                break;
+            default:
+                visibilityIcon = 0;
+        }
+
+        if (visibilityIcon == 0) {
+            return;
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/adapter/StatusDetailedViewHolder.java]: you could just `default: return` above and save these three lines

**Comment by wbrawner** [comment]
Fixed. Sorry for the delay, been a bit busy lately :)

**Comment by connyduck** [comment]
thx!

**Comment by kyori19** [comment]
I think this is mistake.
It should be `ic_lock_outline`, not `ic_lock_open`.

https://github.com/tuskyapp/Tusky/blob/1c34d21a2327e0bf7111ec0b04fcc74d37efa1b5/app/src/main/java/com/keylesspalace/tusky/adapter/StatusDetailedViewHolder.java#L175

**Comment by connyduck** [comment]
Yes you are right @kyori19 
Would you mind to send a pull request?

**Pull request merged**